tests/analyzer/metadata/test_unclaimed_maintainer_email_domain.py

```diff
@@ -0,0 +1,26 @@
+import json
+import os
+import pathlib
+
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
+
+from guarddog.analyzer.metadata.pypi import PypiUnclaimedMaintainerEmailDomainDetector
+from tests.analyzer.metadata.resources.sample_project_info import PYPI_PACKAGE_INFO
+
+with open(os.path.join(pathlib.Path(__file__).parent.resolve(), "resources", "npm_data.json"), "r") as file:
+    NPM_PACKAGE_INFO = json.load(file)
+
+pypi_detector = PypiUnclaimedMaintainerEmailDomainDetector()
+
+
+class TestUnclaimedMaintainerEmailDomain:
+    def test_email_domain_doesnt_exist(self):
```

Comment:
would be great to have an unit test on non-detected packages too

Reply:
Done in 4a161d3